src/SwipeProfiles.js

```diff
@@ -0,0 +1,193 @@
+      var formData = new FormData();
+      formData.append("userId", id)
+      const myRequest = new Request('http://127.0.0.1:5000/getPotentialFriends', {
+            method: 'POST',
+            body: formData,
+          });
+      fetch(myRequest)
+        .then(res => res.json())
+        .then(res => { 
+          if(res.response === "Success"){
+            var potentialFriendsList = res.potentialListId;
+            const displayProfileId = potentialFriendsList.pop();
+            console.log(displayProfileId);
+            this.setState({
+              potentialFriends: potentialFriendsList,
+              displayedUserId: displayProfileId,
+              error: '',
+            });
+
+            this.displayProfile();
+          }        
+          else{      
+            this.setState({
+              error: res.response
+            });
+        }
+        })
+        .catch((error) => {
+            this.setState({
+              error: "Error connecting to backend"
+            });
+          }
+        );
+
+    }
+
+
+
+
+    displayProfile(){
+        const displayId = this.state.displayedUserId;
+        if(displayId) {
+        var formData = new FormData();
+        formData.append("userId", displayId);
+        const myRequest = new Request('http://127.0.0.1:5000/displayProfile', {
+              method: 'POST',
+              body: formData,
+            });
+        fetch(myRequest)
+          .then(res => res.json())
+          .then(res => { 
+            if(res.response === "Success"){
+              this.setState({
+                age: res.age,
+                firstName: res.firstName,
+                description: res.description,
+                interests: res.interests,
+                gender: res.gender,
+                workplace: res.workPlace,
+                error: '',
+              });
+
+              console.log(this.state);
+            } 
+            else{             
+              this.setState({
+                error: res.response
+              });
+          }
+          })
+          .catch((error) => {
+              this.setState({
+                error: "Error connecting to backend"
+              });
+            }
+          );
+        }
+        else {
+          this.setState({
+            error: "There are no potential friends for you"
+          });
+        }
+
+    }    
+    
+    
+    
+    handleSwipe(choice) {
+        console.log(choice);
+        const displayId = this.state.displayedUserId;
+        const currentUserId = this.props?.location?.state?.id;
+        var formData = new FormData();
+        formData.append("currentUserId", currentUserId);
+        formData.append("shownUserId", displayId);
+        formData.append("match", choice);
+        const myRequest = new Request('http://127.0.0.1:5000/swipe', {
+              method: 'POST',
+              body: formData,
+            });
+        fetch(myRequest)
```

Comment:
is this fetching data from back-end every time the user swipes? if yes why do we still wanna store the potential list since we are only looking at one user?